subsys/net/lib/tftp/tftp_client.c

```diff
@@ -0,0 +1,524 @@
+			 */
+			stat = tftpc_recv_data(sock);
+
+			/* Lets check if we were able to get response from the
+			 * server and if the response was Ok or not.
+			 */
+			if (stat > 0) {
+
+				/* Ok - We were able to get response of our read
+				 * request from the TFTP Server. Lets check and
+				 * see what the TFTP Server has to say about our
+				 * request.
+				 */
+				server_response =
+						extract_u16(tftpc_request_buffer);
+
+				/* Did we get some err? */
+				if (server_response == ERROR_OPCODE) {
+
+					/* The server responded with some errors
+					 * here. Lets get to know about the specific
+					 * error and log it. Nothing else we can do
+					 * here really and so should exit.
+					 */
+					LOG_ERR("tftp_get failure - Server returned: %d",
+							extract_u16(tftpc_request_buffer + 2));
+
+					break;
+				}
+
+				/* Did we get some data? */
+				if (server_response == DATA_OPCODE) {
+
+					/* Good News - TFTP Server responded with data.
+					 * Lets talk to the server and get all data.
+					 */
+					LOG_DBG("tftp_get success - Server returned: %d",
+							 extract_u16(tftpc_request_buffer + 2));
+
+					break;
+				}
+
+				/* Did we get some data? */
+				if (server_response == ACK_OPCODE) {
+
+					/* Good News - TFTP Server acked our request. */
+					LOG_DBG("tftp_get success - Server returned: %d",
+							extract_u16(tftpc_request_buffer + 2));
+
+					break;
+				}
+			}
+		}
+
+		/* No of times we had to re-tx this "request". */
+		no_of_retransmists++;
+
+		/* Log this out. */
+		LOG_DBG("tftp_send_request was either unable to get data \
+				 from the TFTP Server.");
+		LOG_DBG("Or failed to get any valid data.");
+		LOG_INF("no_of_retransmists = %d", no_of_retransmists);
+		LOG_INF("Are we re-transmitting: %s",
+				 (no_of_retransmists < TFTP_REQ_RETX) ?
+				 "yes" : "no");
+
+	} while (no_of_retransmists < TFTP_REQ_RETX);
+
+	/* Status? */
+	return server_response;
+}
+
+/* Name: tftp_connect
+ * Description: This function connects with the TFTP Server.
+ */
+static inline s8_t tftp_connect(s32_t sock, struct sockaddr_in *server)
+{
+	/* If the server information is not provided by the user, we
+	 * have to use the default server.
+	 */
+	if (server == NULL) {
+
+		/* Have to use the default server. Populate the IP address. */
+		inet_pton(AF_INET, TFTPC_DEF_SERVER_IP, &def_srv.sin_addr);
+
+		/* Update the pointer. */
+		server = &def_srv;
+	}
+
+	/* Connect with the TFTP Server. */
+	return connect(sock, (struct sockaddr *) server, sizeof(struct sockaddr_in));
+}
+
+/* Name: tftp_get
+ * Description: This function gets "file" from the remote server.
+ */
+int tftp_get(struct sockaddr_in *server, struct tftpc *client,
+			 const char *remote_file, const char *mode)
+{
+
+	s32_t   stat               = TFTPC_UNKNOWN_FAILURE;
+	s32_t   sock;
+	u16_t   server_response;
+	u8_t    no_of_retransmists = 0;
+
+	/* Re-init the global "block number" variable. */
+	tftpc_block_no = 1;
+	tftpc_index    = 0;
```

Comment:
What about `readfds`, `fdnum`, we should initialize them as well.

Reply:
All gone now as we're using `poll` instead of `select`..